index.channel.ts

```diff
@@ -0,0 +1,441 @@
+import { EventEmitter2 } from '@nestjs/event-emitter';
+import {
+  ActionRowBuilder,
+  AttachmentBuilder,
+  ButtonBuilder,
+  ButtonComponent,
+  ButtonInteraction,
+  ButtonStyle,
+  ChannelType,
+  ChatInputCommandInteraction,
+  EmbedBuilder,
+  Events,
+  Message,
+  MessageActionRowComponentBuilder,
+  TextChannel,
+} from 'discord.js';
+import { Request, Response } from 'express';
+import { I18nService } from 'nestjs-i18n';
+import fetch from 'node-fetch';
+
+import { AttachmentService } from '@/attachment/services/attachment.service';
+import { ChannelService } from '@/channel/channel.service';
+import ChannelHandler from '@/channel/lib/Handler';
+import { SubscriberCreateDto } from '@/chat/dto/subscriber.dto';
+import {
+  OutgoingMessageFormat,
+  StdOutgoingAttachmentMessage,
+  StdOutgoingButtonsMessage,
+  StdOutgoingEnvelope,
+  StdOutgoingListMessage,
+  StdOutgoingQuickRepliesMessage,
+  StdOutgoingTextMessage,
+} from '@/chat/schemas/types/message';
+import { BlockOptions } from '@/chat/schemas/types/options';
+import { LabelService } from '@/chat/services/label.service';
+import { MessageService } from '@/chat/services/message.service';
+import { SubscriberService } from '@/chat/services/subscriber.service';
+import { MenuService } from '@/cms/services/menu.service';
+import { LanguageService } from '@/i18n/services/language.service';
+import { LoggerService } from '@/logger/logger.service';
+import { SettingService } from '@/setting/services/setting.service';
+import { SocketRequest } from '@/websocket/utils/socket-request';
+import { SocketResponse } from '@/websocket/utils/socket-response';
+
+import { DiscordBotService } from './discord-api';
+import { DISCORD_CHANNEL_NAME } from './settings';
+import DiscordEventWrapper from './wrapper';
+
+@Injectable()
+export class DiscordChannelHandler extends ChannelHandler<
+  typeof DISCORD_CHANNEL_NAME
+> {
+  private discordBotService: DiscordBotService;
+
+  constructor(
+    settingService: SettingService,
+    channelService: ChannelService,
+    logger: LoggerService,
+    protected readonly eventEmitter: EventEmitter2,
+    protected readonly i18n: I18nService,
+    protected readonly languageService: LanguageService,
+    protected readonly subscriberService: SubscriberService,
+    protected readonly attachmentService: AttachmentService,
+    protected readonly messageService: MessageService,
+    protected readonly menuService: MenuService,
+    protected readonly labelService: LabelService,
+    protected readonly httpService: HttpService,
+  ) {
+    super(DISCORD_CHANNEL_NAME, settingService, channelService, logger);
+  }
+
+  getPath(): string {
+    return __dirname;
+  }
+
+  getChannel(): typeof DISCORD_CHANNEL_NAME {
+    return DISCORD_CHANNEL_NAME;
+  }
+
+  async init(): Promise<void> {
+    try {
+      this.logger.debug('Discord Channel Handler : initialization ...');
+      const { bot_token, app_id } = await this.getSettings();
```

Comment:
We should listen to the change on the settings and update the discordBotService accordingly I think. Example : 

```
/**
   * Updates the access token for the Slack API
   *
   * @param setting
   */
  @OnEvent('hook:slack_channel:access_token')
  async updateAccessToken(setting: THydratedDocument<Setting>) {
    this.api.setAccessToken(setting.value);
  }

  /**
   * Updates the signing secret for the Slack API
   *
   * @param setting
   */
  @OnEvent('hook:slack_channel:signing_secret')
  async updateSigningSecret(setting: THydratedDocument<Setting>) {
    this.api.setSigningSecret(setting.value);
  }

```